ref(querybuilder): Remove ParamsType from tasks

wmak opened the pull request:
- This removes the usage of ParamsType from tasks in favour of SnubaParams
- Added stats_period as an option to SnubaParams as a way to define start and end
- Added a filter_params property on SnubaParams to make this switchover easy so I'm impacting less code (eg. don't have to change the params references in `format_top_events_timeseries_results` even though `params={}` now)

wmak committed:
ref(querybuilder): Remove ParamsType from tasks

- This removes the usage of ParamsType from tasks in favour of
  SnubaParams
- Added stats_period as an option to SnubaParams as a way to define
  start and end

wmak commented on src/sentry/tasks/statistical_detectors.py:
```diff
@@ -924,16 +924,14 @@ def query_functions_timeseries(
-        "project_id": project_ids,
-        "project_objects": projects,
-    }
+    params = SnubaParams(
+        start=end - timedelta(days=14),
+        end=end,
+        projects=projects,
+    )
```
Didn't use `SnubaParams(stats_period=` here since the existing functionality sets end to 1hr in the future

Zylphrex commented on src/sentry/search/events/builder/discover.py:
```diff
@@ -275,6 +278,7 @@ def __init__(
         params: ParamsType,
         interval: int,
         top_events: list[dict[str, Any]],
+        snuba_params: SnubaParams | None = None,
```
Should this be passed to `super().__init__`?

wmak replied:
woops good catch

Zylphrex commented on src/sentry/search/events/types.py:
```diff
@@ -74,24 +75,34 @@ class EventsResponse(TypedDict):
+    def parse_stats_period(self) -> None:
+        if self.stats_period is not None:
+            self.end = django_timezone.now()
+            from sentry.api.utils import get_datetime_from_stats_period
```
Does this import have to be in here?

Zylphrex commented on src/sentry/search/events/types.py:
```diff
@@ -74,24 +75,34 @@ class EventsResponse(TypedDict):
+            self.end = django_timezone.now()
+            from sentry.api.utils import get_datetime_from_stats_period
+
+            self.start = get_datetime_from_stats_period(self.stats_period, self.end)
```
Nit: Thoughts on using `get_date_range_from_stats_period` that doesnt require you to set `self.end` here?

wmak replied:
🤔 those require passing a params object no? I rather not make this more circular than it already is

wmak committed:
ref: address pr comments

- Added automatic self.organization grabbing which the ParamsType ->
  SnubaParams converter was doing automatically
